modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -274,39 +283,36 @@ public String modelFileFolder() {
 
     @Override
     public String toParamName(String name) {
-        // sanitize name
         name = sanitizeName(name, "[^\\w$]");
 
         if ("_".equals(name)) {
             name = "_u";
         }
 
-        // if it's all uppper case, do nothing
-        if (name.matches("^[A-Z_]*$")) {
-            return name;
-        }
-
-        name = getNameUsingModelPropertyNaming(name);
-
-        // for reserved word or word starting with number, append _
-        if (isReservedWord(name) || name.matches("^\\d.*")) {
-            name = escapeReservedWord(name);
-        }
+        name = camelize(name, true);
+        name = toSafeIdentifier(name);
 
         return name;
     }
 
     @Override
     public String toVarName(String name) {
-        name = this.toParamName(name);
-        
-        // if the property name has any breaking characters such as :, ;, . etc.
-        // then wrap the name within single quotes.
-        // my:interface:property: string; => 'my:interface:property': string;
-        if (propertyHasBreakingCharacters(name)) {
```

Comment:
this would never happen, because at this point `name` has been sanitized with even more restrictive rules